lighthouse-cli/test/fixtures/perf/delayed-element.js

```diff
@@ -6,6 +6,24 @@
 
 /* eslint-disable */
 
+/** Create long tasks on the main thread. */
+function stall(ms) {
+  const start = performance.now();
+  while (performance.now() - start < ms) ;
+}
+
+/** Render large number of elements to fill up the backend node cache. */
+function rerender(iterations) {
+  const rerender_ = (i, resolve) => {
+    const filler = `<div>Filler element</div>`.repeat(4000);
+    document.body.innerHTML = `<div id="div-${i}">${i} left</div>${filler}`;
+    if (i === 0) resolve();
+    if (i > 0) requestAnimationFrame(() => rerender_(i - 1, resolve));
+  };
+
+  return new Promise(resolve => rerender_(iterations, resolve));
+}
```

Comment:
Took me a minute to understand but this is really good!

Reply:
thanks @Beytoven! @connorjclark is right though, it was mean of me to expect others to parse a recursive version of something trivially iterative in nature, sorry you had to do that :) hopefully the new version is clearer 👍